### PR TITLE
Upgrade ashwat

### PR DESCRIPTION
**Pull request by galanCA**
## Description

Updates ashwat.cpp for all platforms. The changes performed were:
- Replaces `vsprintf_s` with `vsnprintf`. It seems msvc substitutes its own versions of the `_s` functions, which makes cross-platform support difficult. The `n` provides protection against corruption by ensuring the `number of elements` does not exceed the container size.
- Remove `cdecl` for systems that wont use it. [Link](https://stackoverflow.com/questions/4986844/what-does-cdecl-stand-for#:~:text=The%20term%20CDECL%20originates%20from%20Microsoft%27s%20BASIC%20and%20their%20Mixed%20Language%20Programming%20ecosystem.) containing an explanation for their use. Since they were used in the pass to differentiate the declaration of C vs COBOL vs FORTRAN in the past, the keyword cdecl should not be used currently in this code base as everything should have a C (or C++) declaration.

## Author Progress Checklist:

- [x] Open draft pull request
    - [x] Make title clearly understandable in a standalone change log context
    - [x] Assign yourself the issue
    - [x] Add at least one label (enhancement, bug, or maintenance)
    - [x] Link the issue(s) addressed by this PR (under "Development" in the sidebar menu) 
- [x] Make code changes (if you haven't already)
- [x] Self-review of code
    - [x] My code follows the style guidelines of this project
    - [x] I have added comments to my code, particularly in hard-to-understand areas
    - [x] I have only committed the necessary changes for this fix or feature
    - [x] I have made corresponding changes to the documentation
    - [x] My changes generate no new warnings
    - [x] I have ensured that my fix is effective or that my feature works as intended by:
        - [x] exercising the code changes in the test framework, and/or
        - [x] manually verifying the changes (as explained in the the pull request description above)
    - [x] My changes pass all local tests
    - [x] My changes successfully passes CI checks
    - [x] Add any unit test for proof and documentation.
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
- [ ] Move pull request out of draft mode and assign reviewers
- [ ] Iterate with reviewers until all changes are approved
    - [ ] Make changes in response to reviewer comments
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
    - [ ] Re-request review in GitHub

## Reviewer Checklist:

 - [ ] Read the pull request description
 - [ ] Perform a code review on GitHub
 - [ ] Confirm all CI checks pass and there are no build warnings
 - [ ] Pull, build, and run automated tests locally
 - [ ] Perform manual tests of the fix or feature locally
 - [ ] Add any review comments, if applicable
 - [ ] Submit review in GitHub as either
     - [ ] Request changes, or
     - [ ] Approve
 - [ ] Iterate with author until all changes are approved

### CODE REVIEW

**Comment by nealkruis** [comment]
@galanCA is this ready for review?

File: src/ashwat.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #include "cnglob.h"
 
 #include "ashwat.h"
+#include <climits>
```
Comment (tanaya-bigladder) [inline src/ashwat.cpp:23]: I didn't spot any place where this is being used? 
By the way, you're "adding climits header," not "adding climits library." :)

**Review by tanaya-bigladder — CHANGES_REQUESTED** [review]

**Comment by spahrenk** [comment]
Changed to vsnprintf and merged `main` into `ashwat-update`.

**Comment by nealkruis** [comment]
Superseded by other PRs.

**Pull request closed without merging**